playbooks/roles/backups/files/backup.py

```diff
@@ -338,6 +368,9 @@ def _main():
             elif provider == 's3':
                 upload_to_s3(backup_path, bucket, aws_access_key_id=s3_id,
                              aws_secret_access_key=s3_key)
+            elif provider == 'azure':
+                upload_to_azure_storage(backup_path, bucket, azure_account,
+                                        azure_key)
```

Comment:
improvement for later....rather than having elifs all around we could do something like this:

    cmd_map = {
        'azure': upload_to_azure_storage
        's3': upload_to_s3
       'whateverelse': upload_to_whatever
    }
    cmd_map[provider](arg1, arg2...)

Perhaps something to be done in a larger refactoring already mentioned.